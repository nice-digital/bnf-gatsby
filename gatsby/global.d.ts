// Let TypeScript know we're importing SCSS (and SCSS modules)
// Avoids "Can't import CSS/SCSS modules. TypeScript says “Cannot Find Module”"
// See https://stackoverflow.com/a/41946697/486434
declare module "*.module.scss" {
	const content: { [className: string]: string };
	export default content;
}

// So that we can import JSON files in TypeScript.
declare module "*.json" {
	const value: unknown;

	export default value;
}
