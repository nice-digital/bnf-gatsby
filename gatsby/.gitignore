node_modules/
.cache/
public

# dotenv environment variables file
.env
.env.*
