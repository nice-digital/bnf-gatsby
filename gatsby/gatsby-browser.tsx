import {
	ShouldUpdateScrollArgs,
	type RouteUpdateArgs,
	type WrapPageElementBrowserArgs,
} from "gatsby";
import { type ReactElement } from "react";

import { Layout } from "@/components/Layout/Layout";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
	// Default return statement when prevLocation is falsy
	return;
};

export const wrapPageElement = ({
	element,
	props,
}: WrapPageElementBrowserArgs): ReactElement => (
	<Layout {...props}>{element}</Layout>
);

/**
 * Gatsby hook for overriding scroll position
 * See https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll
 */
export const shouldUpdateScroll = ({
	routerProps: { location },
	prevRouterProps,
	getSavedScrollPosition,
}: ShouldUpdateScrollArgs): boolean | string | [number, number] => {
	if (
		// If there's no previous route props we're coming from an external site, which means
		// we want to scroll to the hash (if there is one), and _not_ a saved scroll position
		!prevRouterProps ||
		// Or we're linking to as hash within the same page
		(prevRouterProps.location.pathname === location.pathname && location.hash)
	) {
		// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = location.hash
					? document.querySelector(location.hash)
					: null;

				if (targetElement) {
					targetElement.setAttribute("tabIndex", "-1");
					(targetElement as HTMLElement).focus();
					targetElement.scrollIntoView();
				}
			});
		});
		return false;
	}

	const savedScrollY = (getSavedScrollPosition(location)?.[1] || 0) as number;
	if (savedScrollY > 0) {
		window.scrollTo(0, savedScrollY);
		return false;
	}

	if (location.hash) {
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = document.querySelector(location.hash);

				// Default to Gatsby's default behaviour if the element doesn't exist
				if (!targetElement) return true;

				// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
				targetElement.setAttribute("tabIndex", "-1");
				(targetElement as HTMLElement).focus();
				targetElement.scrollIntoView();
				return false;
			});
		});
		return false;
	}

	requestAnimationFrame(() => {
		requestAnimationFrame(() => {
			const contentStartElement = document.getElementById("content-start");

			if (!contentStartElement) return true;

			contentStartElement.setAttribute("tabIndex", "-1");

			//scrolling to 0,0 to preserve existing BNF navigation behaviour
			window.scrollTo(0, 0);
			return false;
		});
	});
	// Default to scrolling to the content start element as the standard navigation behaviour

	return true;
};
