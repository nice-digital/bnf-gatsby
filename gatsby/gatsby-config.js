require("dotenv").config({
	path: `.env.${process.env.NODE_ENV}`,
});

const isBNF = process.env.GATSBY_SITE === "bnf";

if (!process.env.GATSBY_SEARCH_URL)
	throw new Error(
		"Env var GATSBY_SEARCH_URL isn't set. Did you forget to add it?"
	);

module.exports = {
	jsxRuntime: "automatic",
	siteMetadata: {
		isBNF,
		siteUrl: `https://${isBNF ? "bnf" : "bnfc"}.nice.org.uk`,
		siteTitleShort: isBNF ? "BNF" : "BNFC",
		siteTitleLong:
			"British National Formulary" + (isBNF ? "" : " for Children"),
	},
	plugins: [
		{
			resolve: `gatsby-source-bnf`,
			options: {
				feedURL: process.env.FEED_URL,
				userKey: process.env.FEED_USER_KEY,
				site: isBNF ? "bnf" : "bnfc",
			},
		},
		// Avoid errors like "ModuleNotFoundError: Module not found: Error: Can't resolve '@/components/Layout/Layout'" when using custom paths in tsconfig.json
		`gatsby-plugin-tsconfig-paths`,
		{
			resolve: `gatsby-plugin-sass`,
			options: {
				// Make CSS modules play nicely with TypeScript, see also global.d.ts for *.scss TS def
				cssLoaderOptions: {
					esModule: false,
					modules: {
						namedExport: false,
					},
				},
			},
		},
		"gatsby-plugin-catch-links",
		"gatsby-plugin-react-helmet",
		"gatsby-plugin-remove-generator",
		{
			resolve: `gatsby-plugin-sitemap`,
			options: {
				excludes: [`/search/`],
			},
		},
		{
			resolve: "gatsby-plugin-google-tagmanager",
			options: {
				id: "GTM-5H5L9GK",
				includeInDevelopment: true,
			},
		},
		{
			resolve: "gatsby-plugin-manifest",
			options: {
				icon: "src/images/icon.png",
			},
		},
	],
};
