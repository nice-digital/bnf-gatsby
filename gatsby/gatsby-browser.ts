import { RouteUpdateArgs } from "gatsby";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
};
