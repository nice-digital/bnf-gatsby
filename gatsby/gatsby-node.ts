import { CreateSchemaCustomizationArgs, CreateWebpackConfigArgs } from "gatsby";

import { initialFieldExtension } from "./src/field-extensions/initial";

/**
 * Gatsby hook for customizing the schema.
 * See https://www.gatsbyjs.org/docs/schema-customization/
 */
export const createSchemaCustomization = ({
	actions: { createFieldExtension, createTypes },
}: CreateSchemaCustomizationArgs): void => {
	createFieldExtension(initialFieldExtension);

	// Add initials for types used in A-Z lists, as it's easier for grouping
	// We don't add the initial field in the source plugin because grouping is a presentation concern
	const typeDefs = `
	  type BnfDrug implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfTreatmentSummary implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfInteractant implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	`;
	createTypes(typeDefs);
};

/**
 * Gatsby hook for overriding webpack config
 * See https://www.gatsbyjs.com/docs/how-to/custom-configuration/add-custom-webpack-config/
 */
export const onCreateWebpackConfig = ({
	stage,
	actions,
	getConfig,
	plugins,
}: CreateWebpackConfigArgs): void => {
	// Disable order warnings from mini CSS plugin
	// See https://github.com/gatsbyjs/gatsby/discussions/30169#discussioncomment-621285
	if (stage === "build-javascript" || stage === "develop") {
		const config = getConfig();

		const miniCss = config.plugins.find(
			(plugin: () => void) => plugin.constructor.name === "MiniCssExtractPlugin"
		);

		if (miniCss) {
			miniCss.options.ignoreOrder = true;
		}

		actions.replaceWebpackConfig(config);

		actions.setWebpackConfig({
			plugins: [plugins.provide({ process: "process/browser" })],
		});
	}
};
