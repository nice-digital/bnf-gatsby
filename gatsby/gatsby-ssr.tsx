import { PreRenderHTMLArgs } from "gatsby";
import React from "react";

export const onPreRenderHTML = ({
	getHeadComponents,
	replaceHeadComponents,
}: PreRenderHTMLArgs): void => {
	const components = getHeadComponents();

	components.push(
		<script
			src={process.env.GATSBY_COOKIE_BANNER_URL}
			type="text/javascript"
			key="cookie-banner"
			async
		></script>
	);

	replaceHeadComponents(components);
};
