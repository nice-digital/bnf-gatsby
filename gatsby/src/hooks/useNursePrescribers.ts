import { useStaticQuery, graphql } from "gatsby";

import { SlugAndTitle, type MenuPageLink } from "@/utils";

export interface NursePrescribers {
	allBnfNursePrescribersFormularyIntroduction: {
		introduction: SlugAndTitle[];
	};
	allBnfNursePrescribersFormularyTreatmentSummary: {
		summaries: SlugAndTitle[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/nurse-prescribers-formulary/${slug}/`,
});

/**
 * Gets the lists of pages for the nurse prescribers formulary by combining:
 * - treatment summaries
 * - introduction
 * from the feed.
 *
 * @returns both the consolidated list for the details page menu, and two separate lists for the index page.
 */
export const useNursePrescribers = (): {
	menuList: MenuPageLink[];
	aboutList: MenuPageLink[];
	treatmentSummariesList: MenuPageLink[];
} => {
	const {
		allBnfNursePrescribersFormularyIntroduction,
		allBnfNursePrescribersFormularyTreatmentSummary,
	} = useStaticQuery<NursePrescribers>(
		graphql`
			query NursePrescribers {
				allBnfNursePrescribersFormularyIntroduction {
					introduction: nodes {
						slug
						title
					}
				}
				allBnfNursePrescribersFormularyTreatmentSummary {
					summaries: nodes {
						slug
						title
					}
				}
			}
		`
	);

	const menuList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries.map(
			slugToHref
		),
		...allBnfNursePrescribersFormularyIntroduction.introduction.map(slugToHref),
	];

	const aboutList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries
			.filter((summary) => summary.slug == "general-guidance")
			.map(slugToHref),
		...allBnfNursePrescribersFormularyIntroduction.introduction.map(slugToHref),
	];

	const treatmentSummariesList = [
		...allBnfNursePrescribersFormularyTreatmentSummary.summaries.map(
			slugToHref
		),
	];

	return { menuList, aboutList, treatmentSummariesList };
};
