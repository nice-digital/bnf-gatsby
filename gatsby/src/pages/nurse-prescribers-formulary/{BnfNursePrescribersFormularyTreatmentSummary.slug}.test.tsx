import { useLocation } from "@reach/router";
import { render, screen, waitFor } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import { mockNursePrescribersPagesQueryData } from "@/hooks/useNursePrescribers.test";

import NursePrescribersFormularyTreatmentSummaryPage, {
	type NursePrescribersFormularyTreatmentSummaryPageProps,
	query,
} from "./{BnfNursePrescribersFormularyTreatmentSummary.slug}";

const pageProps: NursePrescribersFormularyTreatmentSummaryPageProps = {
	data: {
		bnfNursePrescribersFormularyTreatmentSummary: {
			title: "Analgesics",
			slug: "analgesics",
			sections: [
				{
					content: "<p>Analgesic content</p>",
					slug: "analgesic-1",
					title: "Analgesic 1",
				},
				{
					content: "<p>Analgesic content 2</p>",
					slug: "analgesic-2",
					title: "Analgesic 2",
				},
			],
		},
	},
	location: {
		pathname: "/nurse-prescribers-formulary/analgesics/",
	},
};

describe("NursePrescribersFormularyTreatmentSummaryPage", () => {
	beforeEach(() => {
		(useLocation as jest.Mock).mockReturnValue(
			new URL(
				"https://bnf-gatsby-tests.nice.org.uk/nurse-prescribers-formulary/analgesics/"
			)
		);

		(useStaticQuery as jest.Mock).mockReturnValue(
			mockNursePrescribersPagesQueryData
		);
	});

	it("should match snapshot for graphql query", () => {
		expect(query).toMatchSnapshot();
	});

	it("should match snapshot for page contents", () => {
		render(<NursePrescribersFormularyTreatmentSummaryPage {...pageProps} />);

		expect(screen.getByRole("main")).toMatchSnapshot();
	});

	it("should set page title", async () => {
		render(<NursePrescribersFormularyTreatmentSummaryPage {...pageProps} />);

		await waitFor(() => {
			expect(document.title).toStartWith(
				"Analgesics | Nurse Prescribers' Formulary | "
			);
		});
	});

	it("should set meta description for known path on BNF", async () => {
		render(<NursePrescribersFormularyTreatmentSummaryPage {...pageProps} />);

		await waitFor(() => {
			expect(
				// eslint-disable-next-line testing-library/no-node-access
				document.querySelector(`meta[name="description"]`)
			).toHaveAttribute(
				"content",
				"Read non-opioid analgesic prescribing information for nurse prescribers, including aspirin, ibuprofen; and paracetamol for mild-to-moderate pain relief."
			);
		});
	});

	it("should throw error for unknown path", () => {
		expect(() => {
			render(
				<NursePrescribersFormularyTreatmentSummaryPage
					data={{
						bnfNursePrescribersFormularyTreatmentSummary: {
							slug: "unknown",
							title: "Unknown",
							sections: [],
						},
					}}
					location={{
						pathname: "/nurse-prescribers-formulary/unknown/",
					}}
				/>
			);
		}).toThrowError(
			"Couldn't find meta description for page 'Unknown' at path '/nurse-prescribers-formulary/unknown/'. Has the page been added or renamed?"
		);
	});
});
