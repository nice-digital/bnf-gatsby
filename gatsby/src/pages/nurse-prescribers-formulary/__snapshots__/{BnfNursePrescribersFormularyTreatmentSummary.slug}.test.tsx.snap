// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NursePrescribersFormularyTreatmentSummaryPage should match snapshot for graphql query 1`] = `
"
	query ($id: String) {
		bnfNursePrescribersFormularyTreatmentSummary(id: { eq: $id }) {
			title
			slug
			sections {
				...RecordSection
			}
			relatedDrugs {
				title
				slug
			}
		}
	}
"
`;

exports[`NursePrescribersFormularyTreatmentSummaryPage should match snapshot for page contents 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="https://www.nice.org.uk/"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          BNF
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/nurse-prescribers-formulary/"
        >
          Nurse Prescribers' Formulary
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Analgesics
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/nurse-prescribers-formulary/",
        "name": "Nurse Prescribers' Formulary"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Analgesics"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading"
    >
      <span>
        Analgesics
      </span>
    </h1>
  </div>
  <div
    class="grid grid--loose"
    data-testid="body"
  >
    <div
      class="hide-print"
      data-g="12 md:4 lg:3"
    >
       
      <button
        aria-expanded="false"
        aria-label="Expand menu for Nurse Prescribers' Formulary pages"
        class="toggleButton"
        type="button"
      >
        <span>
          Analgesics
        </span>
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
            fill="currentColor"
          />
        </svg>
      </button>
      <nav
        aria-label="Nurse Prescribers' Formulary pages"
        class="stacked-nav"
        id="collapsible-menu"
      >
        <p
          class="stacked-nav__root"
        >
          <a
            aria-current="false"
            href="/nurse-prescribers-formulary/"
          >
            Nurse Prescribers' Formulary
          </a>
        </p>
        <ul
          class="stacked-nav__list"
        >
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="true"
              href="/nurse-prescribers-formulary/analgesics/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Analgesics
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/nurse-prescribers-formulary/appliances-and-reagents-for-diabetes/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appliances and reagents for diabetes
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/nurse-prescribers-formulary/general-guidance/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  General guidance
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/nurse-prescribers-formulary/approved-list-for-prescribing-by-community-practitioner-nurse-prescribers-npf/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Approved list for prescribing by Community Practitioner Nurse Prescribers (NPF)
                </span>
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      data-g="12 md:8 lg:9"
    >
      <div
        class="grid grid--rev grid--loose"
      >
        <div
          data-g="12 lg:3"
        />
        <div
          class="body"
          data-g="12 lg:9"
        >
          <nav
            aria-labelledby="navigate-to-section"
            class="nav sectionNav"
          >
            <h2
              class="heading"
              id="navigate-to-section"
            >
              <span>
                Navigate to section
              </span>
            </h2>
            <ol
              aria-label="Jump links to sections on this page"
              class="linkList"
            >
              <li>
                <a
                  href="#analgesic-1"
                >
                  Analgesic 1
                </a>
              </li>
              <li>
                <a
                  href="#analgesic-2"
                >
                  Analgesic 2
                </a>
              </li>
              <li>
                <a
                  href="#related-drugs"
                >
                  Related drugs
                </a>
              </li>
            </ol>
          </nav>
          <section
            aria-labelledby="analgesic-1"
          >
            <h2
              class="firstHeading"
              id="analgesic-1"
            >
              Analgesic 1
            </h2>
            <div>
              <p>
                Analgesic content
              </p>
            </div>
          </section>
          <section
            aria-labelledby="analgesic-2"
          >
            <h2
              id="analgesic-2"
            >
              Analgesic 2
            </h2>
            <div>
              <p>
                Analgesic content 2
              </p>
            </div>
          </section>
          <section
            aria-labelledby="related-drugs"
          >
            <h2
              id="related-drugs"
            >
              Related drugs
            </h2>
            <ol
              aria-labelledby="related-drugs"
              class="list"
            >
              <li>
                <a
                  class="tag"
                  href="/drugs/aspirin/"
                >
                  Aspirin
                </a>
              </li>
              <li>
                <a
                  class="tag"
                  href="/drugs/ibuprofen/"
                >
                  Ibuprofen
                </a>
              </li>
            </ol>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
